src/test/java/com/google/devtools/build/lib/query2/testutil/AbstractQueryTest.java

```diff
@@ -404,24 +437,42 @@ public void testAttrOperator() throws Exception {
     writeBuildFiles2();
     writeBuildFilesWithConfigurableAttributes();
 
-    assertThat(evalToString("c:*"))
-        .isEqualTo(
-            "//c:BUILD //c:c //c:d //c:d.dwp //c:d.stripped //c:e.cc //c:p //c:q //c:r //c:s");
-    assertThat(evalToString("attr(cmd,':', c:*)")).isEqualTo("//c:c");
-    // Using "empty" pattern will just check existence of the attribute.
-    assertThat(evalToString("attr(cmd,'', c:*)")).isEqualTo("//c:c");
-    assertThat(evalToString("attr(linkshared, 0, c:*)")).isEqualTo("//c:d");
-    assertThat(evalToString("attr('data', 'r', c:*)")).isEqualTo("//c:d");
-    // Empty list attribute value always resolves to '[]'. If list attribute has
-    // more than one value, the will be delimited with ','.
-    assertThat(evalToString("attr('deps', '\\[\\]', c:*)")).isEqualTo("//c:d");
-    assertThat(evalToString("attr('deps', '^..$', c:*)")).isEqualTo("//c:d");
-    assertThat(evalToString("attr('srcs', '\\[[^,]+\\]', c:*)")).isEqualTo("//c:d");
+    if (isWindows()) {
+      assertThat(evalToString("c:*"))
+              .isEqualTo(
+                      "//c:BUILD //c:c //c:d //c:d.dwp //c:d.stripped.exe //c:e.cc //c:p //c:q //c:r //c:s");
+      assertThat(evalToString("attr(cmd,':', c:*)")).isEqualTo("//c:c");
+      // Using "empty" pattern will just check existence of the attribute.
+      assertThat(evalToString("attr(cmd,'', c:*)")).isEqualTo("//c:c");
+      assertThat(evalToString("attr(linkshared, 0, c:*)")).isEqualTo("//c:d");
+      assertThat(evalToString("attr('data', 'r', c:*)")).isEqualTo("//c:d");
+      // Empty list attribute value always resolves to '[]'. If list attribute has
+      // more than one value, the will be delimited with ','.
+      assertThat(evalToString("attr('deps', '\\[\\]', c:*)")).isEqualTo("//c:d");
+      assertThat(evalToString("attr('deps', '^..$', c:*)")).isEqualTo("//c:d");
+      assertThat(evalToString("attr('srcs', '\\[[^,]+\\]', c:*)")).isEqualTo("//c:d");
+
```

Comment:
Can we not duplicated those lines? Instead we can optional add ".exe" based on the OS using Strings.format

Reply:
Okay ill fix that